dali/pipeline/operators/decoder/CMakeLists.txt

```diff
@@ -14,9 +14,17 @@
 
 # Get all the source files and dump test files
 if (BUILD_NVJPEG)
-  list(APPEND DALI_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/nvjpeg_decoder.cu")
+  if (NVJPEG_DECOUPLED_API)
+    list(APPEND DALI_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/nvjpeg_decoder_decoupled_api.cu")
+  else()
+    list(APPEND DALI_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/nvjpeg_decoder.cu")
```

Comment:
Shall we have separate builds for both, regular and decoupled API?
If so, why do we want to maintain regular API?

Reply:
we are building one or the other. The non decoupled API only builds when older nvjpeg is detected.

Reply:
It's just legacy, for people that have the old nvjpeg on their old system